### PR TITLE
[mlir][nfc] Rename @genbool_* as @constant_mask_*

### PR DESCRIPTION
**Pull request by banach-space**
Renames `@genbool_*` tests as `@constant_mask_*`. That's to better
highlight which Op is tested and for better consistency with other test.

In addition,`@genbool_2d` is moved _above_ it's counterparts with
scalable vectors (again, for consistency).

### CODE REVIEW

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir [hunk 1]
```diff
@@ -2685,61 +2685,77 @@ func.func @matrix_ops_index(%A: vector<64xindex>, %B: vector<48xindex>) -> vecto
...
-func.func @genbool_2d_trailing_scalable() -> vector<4x[4]xi1> {
+func.func @constant_mask_2d() -> vector<4x4xi1> {
+  %v = vector.constant_mask [2, 2] : vector<4x4xi1>
+  return %v: vector<4x4xi1>
+}
+
+// CHECK-LABEL: func @constant_mask_2d
+// CHECK: %[[VAL_0:.*]] = arith.constant dense<[true, true, false, false]> : vector<4xi1>
+// CHECK: %[[VAL_1:.*]] = arith.constant dense<false> : vector<4x4xi1>
+// CHECK: %[[VAL_2:.*]] = builtin.unrealized_conversion_cast %[[VAL_1]] : vector<4x4xi1> to !llvm.array<4 x vector<4xi1>>
```
Comment (javedabsar1) [inline mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:2746]: just for my info - why convert-to-llvm results in `unrealized_conversion_cast`? I see it present everywhere in this file so its not a blocker for this diff AFAIU.

Comment (banach-space) [reply mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:2746]: Some things are lowered to a mix of Arith and LLVM, so without `-convert-arith-to-llvm` ... But there will be a lot of other nuances as well that I don't know yet 😅

**Review by dcaballe — APPROVED** [review]
Thanks for the cleanup!

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir (hunk 1 above)
Comment (javedabsar1) [reply mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:2746]: thanks for clarifying.

**Review by javedabsar1 — APPROVED** [review]

**Comment by Saleh5515** [comment]
net::ERR_UNKNOWN_URL_SCHEM

**Pull request merged**